### PR TITLE
[REG-1049] Add ability to customize player replay models

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Adds `isRuntimeObject` to identify assets in the scene that are dynamically spawned
- Converts the ReplayModelManager to a ScriptableObject and supports persistence of the object within the context of the user's project itself
- Adds a button to the replay window as a shortcut to editing the ReplayModelManager asset in the inspector, including a warning text if none have been configured
  - This will auto populate the ReplayModelAssociations any time you load a replay data zip with any missing entity types that aren't static
- Utilizes `isRuntimeObject` flag to know which entities in the replay to automatically add to the list of things needing custom models.
- Updates the Regression Games menu layout options
- Refactors naming to clarify player vs entity from replay data

See also (https://github.com/Regression-Games/RegressionDocs/pull/22)

### CODE REVIEW

File: Runtime/Scripts/RGBotConfigs/RGState.cs [hunk 1]
```diff
@@ -43,6 +47,7 @@ public class RGState: MonoBehaviour
             ["id"] = this.transform.GetInstanceID(),
             ["type"] = objectType,
             ["isPlayer"] = isPlayer,
+            ["isStatic"] = isStatic,
```
Comment (RG-nAmKcAz) [inline Runtime/Scripts/RGBotConfigs/RGState.cs:50]: Maybe we name this `isSceneElement` or `isDynamicallySpawned` ?

Would those names better represent that this is a thing that isn't defined in the scene itself, but it dynamically loaded at runtime based on the game flow , player connection, game event, etc.

Comment (vontell) [reply Runtime/Scripts/RGBotConfigs/RGState.cs:50]: @rcornwall-reg any idea on what developers call this? GameObjects that are dragged into the scene within the editor vs spawned in after the scene starts?

**Review by vontell — APPROVED** [review]
Approving assuming we have an answer for isStatic naming (even if we keep the same name)

File: Editor/Scripts/RGBotReplayWindow.cs [hunk 2]
```diff
@@ -90,7 +90,25 @@ private void OnGUI()
...
+            GUILayout.FlexibleSpace();
+                        
+
             EditorGUILayout.EndHorizontal();
+            
+            GUILayout.Space(5);
+            
+            if (!ReplayModelManager.GetInstance().HasEntries())
+            {
+                EditorGUILayout.HelpBox("Custom Replay Models have not been configured for this project.\nLoad your first replay zip to auto populate the entity types, then configure their model associations using the `Configure Custom Replay Models` button.\n(You can also manually add, edit, or remove associations manually at any time.)", MessageType.Warning, true);
```
Comment (abeizer) [inline Editor/Scripts/RGBotReplayWindow.cs:103]:
```suggestion
                EditorGUILayout.HelpBox("Custom Replay Models have not been configured for this project.\nLoad your first replay zip to auto populate the entity types, then configure their model associations using the `Configure Custom Replay Models` button.\n(You can also manually add, edit, or remove associations at any time.)", MessageType.Warning, true);
```

**Review by abeizer — APPROVED** [review]

**Review by rcornwall-reg — APPROVED** [review]
Looks good to me, I like the scriptable object approach on this

**Comment by rcornwall-reg** [comment]
@RG-nAmKcAz Some suggested names instead of 'isStatic':

'isRuntimeObject'
'isRuntimeGameObject'
'runtimeGameObject'
'runtimeObject'

**Comment by rcornwall-reg** [comment]
@RG-nAmKcAz Approved before the changes, but everything looks good to me 👍

**Pull request merged**